cmd/utils/auth.go

```diff
@@ -32,3 +38,66 @@ func GetCredentials() (*azidentity.ChainedTokenCredential, error) {
 
 	return chain, nil
 }
+
+// delegatedAzureCLICredential is a wrapper around azidentity.AzureCLICredential
+// that will delegate the login to the Azure CLI. This allows 'az login' and
+// 'kubectl-az' to work together on authentication.
```

Comment:
Mmmm not sure. It is indeed simplifying users' life because they don't need to authenticate each time, but at the same time, we are still depending on `az`. Isn't there any way to use the API directly instead of passing through `az`?

Reply:
Oh, but you mentioned an interesting point, users need `az` to get `kubeconfig` anyway.

Reply:
Mmmm maybe we could use this as first step toward the independence from `az`. I think the final aim remains to don't use it at all. I'm not a big fan of calling another binary from the code. 

WDYT?

Reply:
Please consider that it is fine if we need to write the token in a configuration file or something similar. I highly suggest you to try to get some inspiration from the way `az` is doing this.